app/services/search_service.rb

```diff
@@ -44,11 +44,33 @@ def textSearch_all(srchString)
   end
 
   # Search profiles for matching text and package up as a DocResult
-  def textSearch_profiles(srchString)
-    sresult = DocList.new
+  # If order == "recentdesc", search for more recently updated profiles for matching name.
+  # Otherwise, show profiles without any ordering
+  def textSearch_profiles(search_criteria)
+    # filter profiles by username
+    # recent eh o sort
+    # order profiles in desc order
+    # search profiles whre username Like = 'data' order by Node.changed DESC
+
+    user_scope = SrchScope.find_users(search_criteria.query, limit = nil)\
+                          .reorder('') # removing order by id from find_users()
+
+    if search_criteria.order_by == "recent"
+      user_scope = user_scope.joins(:revisions)\
+                             .order("timestamp #{search_criteria.sort_direction}")\
```

Comment:
I guess i have a couple thoughts here. In general, I think it's best not to be adding activeRecord filters or conditions to an object that's had them added at an earlier level of abstraction (like, within SrchScope.find_users), because you can't see them when you're editing, so there's a strong possibility of doing things redundantly or in conflict. Better to do all your ActiveRecord stuff in one place as much as possible. 

Second, I wonder if there's a way to reduce the # of levels of abstraction. Here it seems like we do:

1. A call arrives at `app/api/srch/search.rb` for `get :profiles do`
2. its sent to `app/services/execute_search.rb` for `execute()`
3. that's sent to `app/services/search_service.rb` for `textSearch_profiles()`
4. that's sent to `app/services/srch_scope.rb` for `SrchScope.find_users()`
5. finally an ActiveRecord call is made: `User.search()...`

Each file is now much more readable, but tracing the whole thing is a bit harder. The readability improvements are super, and the files are cleaner and less redundant. Great work! Now, I wonder if we could try flattening the system without losing those advantages? One way to try (and we must admit it's possible every layer is important! But let's try.) is to list why each layer is needed.

1. where incoming calls get routed
2. utility 'execute' function
3. back-end standard service available inside the app
4. scope?
5. do the ActiveRecord call

I'm not saying there is an obvious or easy answer here. And if we stick with 5 layers, that's fine! But this is how I'd recommend we look at the question of "what complexity is necessary" and how readable it is. Make sense? What do you think?

To your narrower question of the optimization, I think it makes sense that you're doing an initial call for users, then making a more complex version of it by joining revisions and then sorting using that new joined table. This looks great! Because the incoming query is already going to be pretty efficient, we think, and then we're just joining something and adding an ordering. How do you like the results it returns?